src/dotty/tools/backend/jvm/CollectSuperCalls.scala

```diff
@@ -0,0 +1,29 @@
+package dotty.tools.backend.jvm
+
+import dotty.tools.dotc.ast.tpd._
+import dotty.tools.dotc.ast.Trees
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+
+class CollectSuperCalls extends MiniPhaseTransform {
```

Comment:
Please add a comment explaining what the purpose of this phase is in some details